### PR TITLE
auto unlock old tasks on project get

### PR DESCRIPTION
**Pull request by feenster**

### CODE REVIEW

File: server/api/project_apis.py [hunk 1]
```diff
@@ -50,10 +49,15 @@ def get(self, project_id):
             error_msg = f'Project GET - unhandled error: {str(e)}'
             current_app.logger.critical(error_msg)
             return {"error": error_msg}, 500
+        finally:
+            try:
```
Comment (hunt3ri) [inline server/api/project_apis.py]: Add a single line comment saying this call will unlock any tasks older than 2 hours

File: server/api/project_apis.py [hunk 2]
```diff
@@ -50,10 +49,15 @@ def get(self, project_id):
...
+                ProjectService.auto_unlock_tasks(project_id)
+            except Exception as e:
+                current_app.logger.critical(str(e))
```
Comment (hunt3ri) [inline server/api/project_apis.py]: Remove one of the blank lines under here, for PEP8 :)

File: server/models/postgis/task.py [hunk 3]
```diff
@@ -105,8 +110,18 @@ def get_last_status(project_id: int, task_id: int):
                     TaskHistory.action == TaskAction.STATE_CHANGE.name) \
             .order_by(TaskHistory.action_date.desc()).first()
 
+        if result == None:
+            return TaskStatus.READY
+
         return TaskStatus[result[0]]
 
+    @staticmethod
+    def get_last_action(project_id: int, task_id: int):
```
Comment (hunt3ri) [inline server/models/postgis/task.py]: Can this method be removed?  Acutally I see this is subtley different from method above, might be worth expaning the comment here to explain

File: server/models/postgis/task.py [hunk 4]
```diff
@@ -179,6 +194,29 @@ def get_tasks(project_id: int, task_ids: List[int]):
...
+             from tasks t,
+                  task_history th
+             where t.project_id = {0}
+             and t.id = th.task_id
+             and t.project_id = th.project_id
+             and t.task_status in (1,3)
+             and th.action in ( 'LOCKED_FOR_VALIDATION','LOCKED_FOR_MAPPING' )
+             and th.action_text is null
+             and th.action_date <= current_timestamp - INTERVAL '2 hours' '''.format(project_id)
+
```
Comment (hunt3ri) [inline server/models/postgis/task.py]: As discussed this should use UTC time from python, if poss

File: server/models/postgis/task.py [hunk 5]
```diff
@@ -179,6 +194,29 @@ def get_tasks(project_id: int, task_ids: List[int]):
...
+        old_tasks = db.engine.execute(old_locks_query)
+
+        if old_tasks.rowcount == 0:
+            return
```
Comment (hunt3ri) [inline server/models/postgis/task.py]: Add a comment here explaining that no tasks older than 2 hours found so return without further processing

File: server/models/postgis/task.py [hunk 6]
```diff
@@ -222,6 +260,16 @@ def lock_task_for_validating(self, user_id: int):
         self.locked_by = user_id
         self.update()
 
+    def clear_task_lock(self):
+        # update the task record
```
Comment (hunt3ri) [inline server/models/postgis/task.py]: Add a docstring under method, explaining purpose is to tidy up after unlokcing

File: server/api/project_apis.py (hunk 2 above)
Comment (feenster) [inline server/api/project_apis.py]: done

File: server/api/project_apis.py (hunk 1 above)
Comment (feenster) [inline server/api/project_apis.py]: done

File: server/models/postgis/task.py (hunk 3 above)
Comment (feenster) [inline server/models/postgis/task.py]: done

File: server/models/postgis/task.py (hunk 5 above)
Comment (feenster) [inline server/models/postgis/task.py]: done

File: server/models/postgis/task.py (hunk 6 above)
Comment (feenster) [inline server/models/postgis/task.py]: Done

File: server/models/postgis/task.py (hunk 4 above)
Comment (feenster) [inline server/models/postgis/task.py]: done

**Pull request merged**